pkgs/tools/typesetting/tectonic/tests.nix

```diff
@@ -0,0 +1,66 @@
+# This package provides `tectonic.passthru.tests`.
+
+{ lib
+, fetchFromGitHub
+, runCommand
+, tectonic
+, curl
+, cacert
+}:
+
+let
+  /*
+    Currently, the test files are only fully available from the `dev` branch of
+    `biber`. If https://github.com/plk/biber/pull/467 is merged and released,
+    we can obtain the test files from `texlive.pkgs.biber.texsource`. For now,
+    we fetch the test files directly from GitHub.
+  */
+  biber-dev-source = fetchFromGitHub {
+    owner = "plk";
+    repo = "biber";
+    rev = "dev";
+    hash = "sha256-GfNV4wbRXHn5qCYhrf3C9JPP1ArLAVSrJF+3iIJmYPI=";
+  };
+  testfiles = "${biber-dev-source}/testfiles";
+
+  notice = builtins.toFile "tectonic-offline-notice" ''
+    # To fetch tectonic's web bundle, the tests require internet access,
+    # which is not available in a build sandbox. To run the tests, try:
+    # `nix-build --no-sandbox --attr tectonic.passthru.tests`
+  '';
```

Comment:
So this test is only meant for manual running? Not by Hydra? I don't think so. How about adding a VM based test? These can access the internet. See examples at `nixos/tests/all-tests.nix`.

Reply:
I spent a night or two looking at the (sparsely documented) NixOS tests, and they are truly amazing beasts! However, it seems that although they do enjoy a network stack, they still do not have internet / "world wide web" access in a sandbox build. Fortunately, I do find out a way to achieve internet access: basically, we make the test as a "fetcher" like `fetchurl`. I have implemented this in #278410.